optimum/habana/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py

```diff
@@ -397,94 +475,251 @@ def __call__(
+                        # compute the previous noisy sample x_t -> x_t-1
+                        latents_dtype = latents_batch.dtype
+                        latents_batch = self.scheduler.step(noise_pred, timestep, latents_batch, return_dict=False)[0]
+
+                        if latents_batch.dtype != latents_dtype:
+                            if torch.backends.mps.is_available():
+                                # some platforms (eg. apple mps) misbehave due to a pytorch bug: https://github.com/pytorch/pytorch/pull/99272
+                                latents_batch = latents_batch.to(latents_dtype)
+
+                        if callback_on_step_end is not None:
+                            print(" in callback")
+                            callback_kwargs = {}
+                            for k in callback_on_step_end_tensor_inputs:
+                                callback_kwargs[k] = locals()[k]
+                            callback_outputs = callback_on_step_end(self, i, timestep, callback_kwargs)
+
+                            latents_batch = callback_outputs.pop("latents", latents_batch)
+
+                            _prompt_embeds = callback_outputs.pop("prompt_embeds", None)
+                            _negative_prompt_embeds = callback_outputs.pop("negative_prompt_embeds", None)
+                            if _prompt_embeds is not None and _negative_prompt_embeds is not None:
+                                text_embeddings_batch = torch.cat([_negative_prompt_embeds, _prompt_embeds])
+                            _pooled_prompt_embeds = callback_outputs.pop("pooled_prompt_embeds", None)
+                            _negative_pooled_prompt_embeds = callback_outputs.pop("negative_pooled_prompt_embeds", None)
+                            if _pooled_prompt_embeds is not None and _negative_pooled_prompt_embeds is not None:
+                                pooled_prompt_embeddings_batch = torch.cat([_negative_pooled_prompt_embeds, _pooled_prompt_embeds])
+
+
+                        # call the callback, if provided
+                        if i == len(timesteps) - 1 or ((i + 1) > num_warmup_steps and (i + 1) % self.scheduler.order == 0):
+                            progress_bar.update()
+                        
+                        hb_profiler.step()
+
+                        if not self.use_hpu_graphs:
+                            htcore.mark_step(sync=True)
+                    
 
                 t1 = warmup_inference_steps_time_adjustment(t1, t1, num_inference_steps, throughput_warmup_steps)
                 speed_metrics_prefix = "generation"
                 speed_measures = speed_metrics(
                     split=speed_metrics_prefix,
                     start_time=t0,
-                    num_samples=num_batches * batch_size,
-                    num_steps=num_batches * batch_size * num_inference_steps,
+                    num_samples= batch_size,  
+                    num_steps= batch_size * num_inference_steps,
```

Comment:
Check total num steps.  When `t1` is captured after warmup, then num of steps would need to depend on `num_batches` and warmup size